cloudinit/distros/photon.py

```diff
@@ -0,0 +1,387 @@
+            resolve_data = rf.readlines()
+        LOG.debug('Old Resolve Data\n')
+        LOG.debug('%s', resolve_data)
+        for item in resolve_data:
+            if ((nameservers and ('DNS=' in item)) or
+                    (searchdomains and ('Domains=' in item))):
+                continue
+            else:
+                new_resolve_data.append(item)
+
+        new_resolve_data = new_resolve_data + \
+            convert_resolv_conf(nameservers, searchdomains)
+        LOG.debug('New resolve data\n')
+        LOG.debug('%s', new_resolve_data)
+        if nameservers or searchdomains:
+            util.write_file(self.resolve_conf_fn, ''.join(new_resolve_data))
+
+        return dev_names
+
+    def _write_interface_file(self, net_fn, net_cfg, route_entry):
+        if not net_cfg['Name']:
+            return
+        content = '[Match]\n'
+        content += 'Name=%s\n' % (net_cfg['Name'])
+        if 'MACAddress' in net_cfg:
+            content += 'MACAddress=%s\n' % (net_cfg['MACAddress'])
+        content += '[Network]\n'
+
+        if 'DHCP' in net_cfg and net_cfg['DHCP'] in {'yes', 'ipv4', 'ipv6'}:
+            content += 'DHCP=%s\n' % (net_cfg['DHCP'])
+        else:
+            if 'Address' in net_cfg:
+                content += 'Address=%s\n' % (net_cfg['Address'])
+            if 'Gateway' in net_cfg:
+                content += 'Gateway=%s\n' % (net_cfg['Gateway'])
+            if 'DHCP' in net_cfg and net_cfg['DHCP'] == 'no':
+                content += 'DHCP=%s\n' % (net_cfg['DHCP'])
+
+            route_index = 0
+            found = True
+            if route_entry:
+                while found:
+                    route_name = 'routes.' + str(route_index)
+                    if route_name in route_entry:
+                        content += '[Route]\n'
+                        if len(route_entry[route_name]) != 2:
+                            continue
+                        content += 'Gateway=%s\n' % (
+                            route_entry[route_name][0])
+                        content += 'Destination=%s\n' % (
+                            route_entry[route_name][1])
+                    else:
+                        found = False
+                    route_index += 1
+
+        util.write_file(net_fn, content)
+
+    def _bring_up_interfaces(self, device_names):
+        cmd = ['systemctl', 'restart', 'systemd-networkd', 'systemd-resolved']
+        LOG.debug('Attempting to run bring up interfaces using command %s',
+                  cmd)
+        try:
+            (_out, err) = subp.subp(cmd)
+            if len(err):
+                LOG.warning('Running %s resulted in stderr output: %s',
+                            cmd, err)
+            return True
+        except subp.subp.ProcessExecutionError:
+            util.logexc(LOG, 'Running network bringup command %s failed', cmd)
+            return False
+
+    def _select_hostname(self, hostname, fqdn):
```

Comment:
I know your logic here is slightly different, but if you set the class variable `prefer_fqdn = True`, this override shouldn't be needed.